### PR TITLE
Use hpack not cabal

### PR DESCRIPTION
**Pull request by ChrisPenner**
Adds *.cabal to gitignore

Generated package.yaml using [`hpack-convert`](https://github.com/yamadapc/hpack-convert)

---

Unfortunately I'm getting some compilation problems related to file path case sensitivity :'(

There's a note on this here;
https://github.com/haskell/cabal/issues/4739

I've tried adding `--ghc-options=-optP-Wno-nonportable-include-path`; which removes the warning but the actual gcc linking fails later on.

Not entirely sure why this is breaking as a result of the switch; seems mostly to be affecting Proxy; so maybe I'll switch the cabal file back for just that one?

Definitely open to thoughts on this PR; is this something we should bother doing at all?

### CODE REVIEW

File: libs/api-client/package.yaml [hunk 1]
```diff
@@ -0,0 +1,102 @@
+name: api-client
+version: '0.4.2'
+synopsis: (Internal) Wire HTTP API Client
+description: An (internal) client library for the Wire HTTP API.
+category: Network
+author: Wire Swiss GmbH
+maintainer: Wire Swiss GmbH <backend@wire.com>
+license: AGPL-3
+default-extensions:
```
Comment (neongreen) [inline libs/api-client/package.yaml]: Let's use https://github.com/sol/hpack#defaults for this section

Comment (ChrisPenner) [reply libs/api-client/package.yaml]: I didn't make any manual edits since that would take a fair amount more time than just using the automated tool; do you think it's worthwhile to dig through and clean everything up? I suppose if we don't do it now it's unlikely to happen. Should I look into somehow adding some sort of yaml inheritance for these sort of settings or do we not care enough to look into that?

Comment (neongreen) [reply libs/api-client/package.yaml]: You don't need YAML inheritance, it's a built-in hpack feature. You can create a file in the repo:

```
-- package-defaults.yaml

default-extensions:
  - ...
```

And then in every package.yaml file you say on the top level:

```
defaults:
  path: package-defaults.yaml
```

Comment (neongreen) [reply libs/api-client/package.yaml]: Oh, or maybe it should be `local: package-defaults.yaml`; I'm not sure.

Comment (fisx) [reply libs/api-client/package.yaml]: @ChrisPenner if you don't to I'm happy to do this.

Comment (ChrisPenner) [reply libs/api-client/package.yaml]: After taking a peak; yup this is a good idea 😄 

I added it.

**Comment by neongreen** [comment]
It doesn't seem that `module-defaults` is a better name, since those are really package defaults and not module defaults (e.g. I can easily imagine `author:        Wire Swiss GmbH` also going there).

Regardless, approved.

File: services/brig/test/resources/turn/servers-v2.txt [hunk 2]
```diff
@@ -1,2 +1 @@
-turn:localhost:3478
-turn:localhost:3479
+turn:localhost:3478
```
Comment (jschaul) [inline services/brig/test/resources/turn/servers-v2.txt]: unrelated change, undo?

File: services/spar/package.yaml [hunk 3]
```diff
@@ -9,7 +11,10 @@ license-file:  LICENSE
 category:      System
 description:   'See README.md'
 
-ghc-options: -j -O2 -Wall -Wno-redundant-constraints -fwarn-tabs -Werror -optP-Wno-nonportable-include-path
+ghc-options:
+- -j
+- -Wno-redundant-constraints
+- -Werror
```
Comment (jschaul) [inline services/spar/package.yaml]: Can you motivate the change (removal) of ghc options here?

**Comment by neongreen** [comment]
-O2 went into module-defaults.yaml.

**Comment by jschaul** [comment]
> -O2 went into module-defaults.yaml.

Ah I see, thanks, I missed that in the rather large diff.

:+1:

File: services/spar/package.yaml (hunk 3 above)
Comment (ChrisPenner) [reply services/spar/package.yaml]: As Artyom mentioned, several of the more pervasive options are now globally applied via `module-defaults.yaml`

File: services/brig/test/resources/turn/servers-v2.txt (hunk 2 above)
Comment (ChrisPenner) [reply services/brig/test/resources/turn/servers-v2.txt]: Very strange, no idea what happened here; great catch!

File: .gitignore [hunk 4]
```diff
@@ -39,7 +39,6 @@ services/nginz/src
 services/.env
 tools/api-simulations/mailboxes.json
 tools/api-simulations/reports
-.DS_Store
```
Comment (ChrisPenner) [inline .gitignore]: This was duplicated

File: .gitignore [hunk 5]
```diff
@@ -61,6 +60,8 @@ spar.integration-aws.yaml
 integration-aws.yaml
 DOCKER_ID*
 swagger-ui
-services/spar/spar.cabal
```
Comment (ChrisPenner) [inline .gitignore]: Now covered by `*.cabal`

File: services/brig/test/resources/turn/servers-v2.txt (hunk 2 above)
Comment (tiago-loureiro) [reply services/brig/test/resources/turn/servers-v2.txt]: Out of curiosity: when running brig's integration tests, one of the tests creates that 2nd entry... but it should always fallback to the original after, something funky happened there

**Comment by ChrisPenner** [comment]
Any have other global GHC options they'd like added before I send this through?
